cfme/tests/test_rest.py

```diff
@@ -517,3 +519,39 @@ def test_edit_requests(self, rest_api, pending_requests, from_detail):
         for request in pending_requests:
             request.reload()
             assert request.options['arbitrary_key_allowed'] == 'test_rest'
+
+    @pytest.mark.uncollectif(lambda: current_version() < '5.7')
+    def test_create_automation_requests_parallel(self, rest_api):
+        """Create automation requests in parallel.
+
+        Metadata:
+            test_flag: rest
+        """
+        output = mp.Queue()
+        entry_point = rest_api._entry_point
+        auth = rest_api._auth
+
+        def _gen_automation_requests(entry_point, auth, output):
+            api = MiqApi(entry_point, auth, verify_ssl=False)
+            requests_data = automation_requests_data(
+                'nonexistent_vm', requests_collection=True, approve=False)
+            api.collections.requests.action.create(*requests_data[:2])
+            result = (api.response.status_code, api.response.json())
+            output.put(result)
+
+        processes = [
+            mp.Process(target=_gen_automation_requests, args=(entry_point, auth, output))
+            for _ in range(4)]
+
+        for proc in processes:
+            proc.start()
+
+        # wait for all processes to finish
+        for proc in processes:
+            proc.join()
```

Comment:
@RonnyPfannschmidt I've spent about an hour trying to make it work with `concurrent.futures` but it seems to just hang, not sure why. I might try to debug it when I have some spare time, right now it seems to me that it's harder to use and debug than the plain multiprocessing module.
```
    def test_create_automation_requests_parallel(self, rest_api):
        entry_point = rest_api._entry_point
        auth = rest_api._auth

        def _gen_automation_requests(body):
            api = MiqApi(entry_point, auth, verify_ssl=False)
            api.collections.requests.action.create(body)
            return (api.response.status_code, api.response.json())

        requests = [automation_requests_data(
            'nonexistent_vm', requests_collection=True, approve=False)[:2] for _ in range(4)]

        with concurrent.futures.ProcessPoolExecutor() as executor:
            for status, response in executor.map(_gen_automation_requests, requests):
                assert status == 200
                for result in response['results']:
                    assert result['request_type'] == 'automation'

```